profiles/schema.py

```diff
@@ -804,6 +807,13 @@ def mutate(parent, info, input):
                 if address.is_empty():
                     address.delete()
 
+        service_client_id = input.pop("service_client_id", None)
+        if service_client_id:
+            service = Service.objects.get(client_ids__client_id=service_client_id)
```

Comment:
I'm still not familiar with graphene, but does if validate the service_client_id? So that this can never raise DoesNotExist?

Reply:
Not sure which part of the code exactly you meant, but I'll try to clarify.

Graphene ensures that the argument type is according to the GraphQL schema. In the schema the type of the `service_client_id` argument is string. So, if `service_client_id` exists in `input`, it's a string (could be an empty string or `None` too, but those are falsy, so the execution doesn't get inside the `if` block).

If you meant the `service = Service.objects.get(client_ids__client_id=service_client_id)` part, yes, that can raise a `DoesNotExist` exception -- and that's actually expected. There's also a test for that, called `test_giving_non_existing_service_client_id_results_in_object_does_not_exist_error`. So that exception is turned into a `OBJECT_DOES_NOT_EXIST_ERROR` error code in the response.

Reply:
Ok. That kind of implicit error handling is a bit strange and non-intuitive. Also there are many changes before that point that I guess don't committed to the database because of the transaction?

Reply:
> Ok. That kind of implicit error handling is a bit strange and non-intuitive.

At first, yes. This same pattern has been used throughout the code base.

> Also there are many changes before that point that I guess don't committed to the database because of the transaction?

Correct. Also a recurring pattern in the code base.